mondrian/src/main/java/mondrian/olap/Property.java

```diff
@@ -755,6 +757,16 @@ public Datatype getType() {
         return type;
     }
 
+    public SqlStatement.Type getSqlType() {
```

Comment:
This method is not needed. If the schema overrides the type, we use it, if not we don't. No guessing.